drivers/hmis/app/models/hmis/ce/referral_enroller.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Hmis::Ce
+  class ReferralEnroller
+    attr_reader :referral
+    def initialize(referral)
+      @referral = referral
+    end
+
+    def create_enrollment(message, user)
+      project = referral.opportunity.project
+      raise 'access denied' unless user.can_enroll_clients_for?(project)
+
+      submitted_values = message.step&.submitted_values&.symbolize_keys
+      coc_code_arg = submitted_values ? submitted_values[:coc_code] : nil
+      coc_code = project.determine_coc_code(coc_code_arg: coc_code_arg)
+
+      # TODO(#7321) - carry over the client's household members from the source enrollment
+
+      enrollment = Hmis::Hud::Enrollment.new(
+        client: referral.client,
+        project: project,
+        entry_date: Date.current,
+        user: Hmis::Hud::User.from_user(user),
+        household_id: Hmis::Hud::Base.generate_uuid,
+        relationship_to_hoh: 1, # Head of Household
+        enrollment_coc: coc_code,
+      )
+      raise 'referral generated invalid enrollment' unless enrollment.valid?
+
+      entry_date_errors = Hmis::Hud::Validators::EnrollmentValidator.validate_entry_date(enrollment)
+      entry_date_errors.reject! { |e| e.warning? && e.type == :information }
+      error_out(entry_date_errors.first.full_message) unless entry_date_errors.empty?
```

Comment:
Hmmm good thought! Raising doesn't seem ideal, but we can probably address it later. It seems like the most relevant validation that might come up is if the client is already enrolled. I'll link related issue https://github.com/greenriver/hmis-warehouse/pull/4526 for when we added a similar validation error to existing referral functionality.

I don't see any reason to raise `warning`-level valdiations.

```suggestion
      # Raise an exception if there are any entry date validation errors, which may mean that the client is already enrolled at the project. Future improvement would be to return this as a validation error instead of raising
      entry_date_errors = Hmis::Hud::Validators::EnrollmentValidator.validate_entry_date(enrollment)
      entry_date_errors.reject! { |e| e.warning? }
      error_out(entry_date_errors.map(&:full_message).join(', ')) unless entry_date_errors.empty?
```